extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -14,8 +15,10 @@ export interface ValidationResult {
 export type Validator = () => Promise<ValidationResult>;
 export type ValidationValueType = string | number | undefined;
 
-export type VariableValueGetter = (variable: string) => Promise<ValidationValueType>;
+export type onValidation = (isValid: boolean) => Thenable<void>;
```

Comment:
does it make more sense to make this an event?

Reply:
not really because we just need a callback and not an event that will support multiple subscriptions.

Reply:
I am thinking a little more about this suggestion because even if we need only a callback now, making it an event could be useful

Reply:
I'd really suggest at this point to get it in as is and then follow up with a separate PR to make that change if you think it's useful (I would agree it is).

This PR is already massive enough and it's starting to get really difficult to keep track of all the changes made in each iteration.

Reply:
ok - sounds good.